packages/vaadin-grid/src/vaadin-grid-row-details-mixin.js

```diff
@@ -140,34 +144,40 @@ export const RowDetailsMixin = (superClass) =>
      * @param {!GridItem} item
      * @protected
      */
-    _toggleDetailsCell(row, item) {
+    _toggleDetailsCell(row, detailsOpened) {
       const cell = row.querySelector('[part~="details-cell"]');
       if (!cell) {
         return;
       }
-      const detailsHidden = !this._isDetailsOpened(item);
-
-      if ((!cell._instance && !cell._renderer) || cell.hidden !== detailsHidden) {
-        cell.hidden = detailsHidden;
-        if (detailsHidden) {
-          row.style.removeProperty('padding-bottom');
-        } else {
-          if (this.rowDetailsRenderer) {
-            cell._renderer = this.rowDetailsRenderer;
-            cell._renderer.call(this, cell._content, this, { index: row.index, item: item });
-          } else if (this._rowDetailsTemplate && !cell._instance) {
-            // Stamp the template
-            cell._instance = this._rowDetailsTemplate.templatizer.createInstance();
-            cell._content.innerHTML = '';
-            cell._content.appendChild(cell._instance.root);
-            this._updateItem(row, item);
-          }
 
-          flush();
-          row.style.setProperty('padding-bottom', `${cell.offsetHeight}px`);
+      cell.hidden = !detailsOpened;
 
-          requestAnimationFrame(() => this.notifyResize());
-        }
+      if (cell.hidden) {
+        return;
+      }
+
+      // Assigns either a renderer or a template when the details cell is opened.
+      // The content of the details cell is rendered later in the `_updateItem` method.
+      if (this.rowDetailsRenderer) {
+        cell._renderer = this.rowDetailsRenderer;
```

Comment:
Probably, it would be better to keep the rendering logic here to guarantee that the details’ cell is toggled and rendered.